### PR TITLE
Check PR Author instead of Action Actor

### LINKED ISSUE #112: Request: Check PR Author Instead of Action Actor
https://github.com/dependabot/fetch-metadata/blob/ffa0846490b6178282839adcc9f50954b247e373/src/dependabot/verified_commits.ts#L20-L24

The above code seems to check whether or not the Actions actor is Dependabot and avoids calling the API if the actor isn't Dependabot.   Can we consider checking the PR author instead of/in addition to the actor?  The reason I ask for this is that there are times where I need to re-run the action and I don't want to have to get dependabot to recreate the PR via `@dependabot recreate`, since that takes more time.

The author is available in the [PR event data](https://docs.github.com/en/rest/reference/pulls) and I'm able to access it in actions yml using `github.event.pull_request.user.login`

Happy to open a PR for this if desired

### PR DESCRIPTION
**Pull request by mwaddell**
closes #112

### CODE REVIEW

File: .github/workflows/dependabot-auto-merge.yml
Comment (brrygrdn) [inline .github/workflows/dependabot-auto-merge.yml:9]: ✨  Thanks for updating the documentation as well

**Review by brrygrdn — APPROVED** [review]
Thanks @mwaddell! This change looks good to me.

**Comment by xt0rted** [comment]
I adjusted [my workflow](https://github.com/xt0rted/.github/blob/main/.github/workflows/dependabot-auto-merge.yml) to take advantage of this change, but now it fails if I add commits to to the PR.

image

Typical reasons for this would be to add an entry to the changelog, fix linting rules, or as happened today fix an incorrect version update (a couple actions used `v2` instead of `v2.0.0`).

What's the best way to work around this? Maybe a new output indicating if non-dependabot commits were found so you could skip subsequent steps?

**Comment by mwaddell** [comment]
@brrygrdn What's the reason for failing the run if the PR contains commits after the original dependabot one?   Is there a good reason not to delete https://github.com/dependabot/fetch-metadata/blob/main/src/dependabot/verified_commits.ts#L35:L38

**Comment by mwaddell** [comment]
> @brrygrdn What's the reason for failing the run if the PR contains commits after the original dependabot one? Is there a good reason not to delete https://github.com/dependabot/fetch-metadata/blob/main/src/dependabot/verified_commits.ts#L35:L38

@brrygrdn - I added #166 to remove those lines in order to address the issue noted by @xt0rted - let me know if there's a reason we can't do this.

**Comment by brrygrdn** [comment]
That's a good question, it's something I've actually been blocked by myself this week. I originally added it as we were trying to facilitate narrowest possible definition of a Dependabot PR, i.e.
- Created by PR
- Exclusively containing code committed by Dependabot

I don't think any potential benefit or risk it mitigates really offsets the fact that an extremely common workflow like merging in the target branch is blocked. I'll have a chat with the team on Monday morning to get a second opinion, but I'm leaning towards removing this.

**Pull request merged**
